addons/web/static/tests/core/utils/timing_tests.js

```diff
@@ -225,6 +225,71 @@ QUnit.module("utils", () => {
                 assert.verifySteps([]);
             });
 
+            QUnit.test(
+                "useDebounced: alwaysApply trigger callBack if the debounced func is called but not resolve before comp destroy",
+                async function (assert) {
+                    const { advanceTime } = mockTimeout();
+                    class C extends Component {
+                        static template = xml`<button class="c" t-on-click="() => this.debounced('hello')">C</button>`;
+                        setup() {
+                            this.debounced = useDebounced(
+                                (p) => assert.step(`debounced: ${p}`),
+                                1000,
+                                {
+                                    alwaysApply: true,
+                                }
+                            );
+                        }
+                    }
+                    const fixture = getFixture();
+                    const comp = await mount(C, fixture);
+                    assert.verifySteps([]);
+                    assert.containsOnce(fixture, "button.c");
+
+                    await click(fixture, "button.c");
+                    await advanceTime(999);
+                    assert.verifySteps([]);
+                    await advanceTime(1);
+                    assert.verifySteps(["debounced: hello"]);
+
+                    await click(fixture, "button.c");
+                    await advanceTime(999);
+                    assert.verifySteps([]);
+                    destroy(comp);
+                    await advanceTime(1);
+                    assert.verifySteps(["debounced: hello"]);
```

Comment:
This check should be done immediately after destroy and the time should not need to be advanced.